integration_test.go

```diff
@@ -2374,3 +2374,45 @@ func TestDebugState(t *testing.T) {
 	assert.Equal(t, 1, len(keyRegionCache.(map[string]interface{})))
 	assert.Equal(t, 1, len(clientRegionCache.(map[string]interface{}))) // only have one client
 }
+
+type regionInfoAndAddr struct {
+	regionInfo hrpc.RegionInfo
+	addr       string
+}
+
+// Test looking up meta table for all regions.
+func TestMetaLookupForTable(t *testing.T) {
```

Comment:
I feel like it would be better if this test just called `CacheRegions` on the client and then somehow verified that all the regions were loaded.

Reply:
This is a bit round-about, but you can call `gohbase.DebugState(c)` to get state about the cache as JSON. You could then parse that json back into maps/structs and check that way.
If that is too cumbersome, I think you could expose a method for getting the cache state.

Reply:
Thanks! I was able to use `DebugState` to check the region cache. I changed the test to check the region cache before and after running `CacheRegions`